chainer/backend.py

```diff
@@ -65,7 +65,8 @@ def copyto(dst, src):
     elif isinstance(dst, cuda.ndarray):
         if isinstance(src, chainer.get_cpu_array_types()):
             src = numpy.asarray(src)
-            if dst.flags.c_contiguous or dst.flags.f_contiguous:
+            if (src.dtype == dst.dtype
```

Comment:
`set` does not accept an array of different dtype, so we choose to fall through if dtypes don't match.